qujax/density_matrix.py

```diff
@@ -215,3 +215,50 @@ def no_params_to_densitytensor_func(densitytensor_in: jnp.ndarray = None) -> jnp
         return no_params_to_densitytensor_func
 
     return params_to_densitytensor_func
+
+
+def densitytensor_to_measurement_probabilities(densitytensor: jnp.ndarray,
+                                               qubit_inds: Sequence[int]) -> jnp.ndarray:
+    """
+    Extract array of measurement probabilities given a densitytensor and some qubit indices to measure
+    (in the computational basis).
+    I.e. the ith element of the array corresponds to the probability of observing the bitstring
+    represented by the integer i on the measured qubits.
+
+    Args:
+        densitytensor: Input densitytensor.
+        qubit_inds: Sequence of qubit indices to measure.
+
+    Returns:
+        Normalised array of measurement probabilities.
+    """
+    n_qubits = densitytensor.ndim // 2
+    n_qubits_measured = len(qubit_inds)
+    qubit_inds_trace_out = [i for i in range(n_qubits) if i not in qubit_inds]
+    return jnp.diag(partial_trace(densitytensor, qubit_inds_trace_out).reshape(2 * n_qubits_measured,
+                                                                               2 * n_qubits_measured)).real
+
+
+def densitytensor_to_measured_densitytensor(densitytensor: jnp.ndarray,
+                                            qubit_inds: Sequence[int],
+                                            measured_int: int) -> jnp.ndarray:
```

Comment:
Would it make more sense for `measured_int` to be a bit string instead?

Reply:
Maybe we could accept either?

And if a bitstring is given, we convert it to an int

Reply:
Sounds reasonable, and that only uses information about the shape, so the function should still be JIT'able